examples/tn_simulation.py

```diff
@@ -0,0 +1,157 @@
+"""
+Usage and application of tensor network simulations
+=====================================================
+
+In this example, we will illustrate the usage and application of TN simulation of MBQC.
+
+Firstly, let's import the relevant modules:
+"""
+
+from functools import reduce
+
+import cotengra as ctg
+import matplotlib.pyplot as plt
+import numpy as np
+import opt_einsum as oe
+from scipy.optimize import minimize
+
+from graphix import Circuit
+
+# This application will be for the QAOA (Quantum Approximate Optimization Algorithm),
+# which is an algorithm that can be used for example to solve combinatorical optimization problems.
+# For this example a max cut problem will be solved on a star-like graph, so we can easier validate the results.
+
+
+# Let's start with defining a helper function for buidling the circuit.
+def ansatz(circuit, n, gamma, beta, iterations):
+    for j in range(0, iterations):
+        for i in range(1, n):
+            circuit.cnot(i, 0)
+            circuit.rz(0, gamma[j])
+            circuit.cnot(i, 0)
+        for i in range(0, n):
+            circuit.rx(i, beta[j])
+
+
+n = 5  # This will result in a graph that would be too large for statevector simulation.
+iterations = 2  # Define the number of iterations in the quantum circuit.
+gamma = 2 * np.pi * np.random.rand(iterations)
+beta = 2 * np.pi * np.random.rand(iterations)
+# Define the circuit.
+circuit = Circuit(n)
+ansatz(circuit, n, gamma, beta, iterations)
+
+# Transpile Circuit into pattern as it is needed for creating the TN.
+pattern = circuit.transpile(opt=True).pattern
+# Optimizing according to standardization algorithm of graphix.
+pattern.standardize()
+pattern.shift_signals()
+
+# Print some properties of the graph.
+nodes, edges = pattern.get_graph()
+print(f"Number of nodes: {len(nodes)}")
+print(f"Number of edges: {len(edges)}")
+
+# Optimizing by performing Pauli measurements in the pattern using efficient stabilizer simulator.
+pattern.perform_pauli_measurements(use_rustworkx=True)
+
+# Simulate using the TN backend of graphix, which will return an MBQCTensorNet object.
+# The graph_prep argument is optional,
+# but with 'parallel' the TensorNetworkBackend will prepeare the graph state faster.
+mbqc_tn = pattern.simulate_pattern(backend="tensornetwork", graph_prep="parallel")
+sv = mbqc_tn.to_statevector().flatten()
+print("Statevector after the simulation:", sv)
+
```

Comment:
here, could you add description of the TN that's constructed, as per issue: `Detailed, step-by-step description of the TN construction and contraction in tensornetwork backend of graphix`? I propose:
- visualize the tensornetwork using https://quimb.readthedocs.io/en/latest/tensor-drawing.html if applicable
- describe how this is constructed, by describing https://github.com/TeamGraphix/graphix/blob/5194a88240b2f6ea11c70d831329774bf2ba7cc4/graphix/sim/tensornet.py#L384 (c.f. https://journals.aps.org/pra/abstract/10.1103/PhysRevA.76.052315).

Reply:
see also https://github.com/jcmgray/quimb/blob/main/docs/tensor-design.ipynb

Reply:
Thank you for the review and the sources! I pushed a commit that aims to address the missing parts you were mentioning, but the description might still be a bit short, so please tell me if it needs further improvement. 
Also, I tested locally and I think the file renders well with sphinx-gallery now.

Reply:
Thank you for the update! I believe just a little more steps for the description, with illustrative code snippet or visualization, would be needed to finalize this PR. I would like to propose:

- `for more information look for properties of MPS`  needs to be more concrete. can you add one or two sentences, and link to appropriate reference (e.g. https://quimb.readthedocs.io/en/latest/tensor-1d.html if applicable?)
- use a simple graph to illustrate what kind of inforamtion is stored (after `For every node a list is created, which stores the index labels for each dimension for that given node.` which now is a little vague).  or perhaps just inspect what's in one of the tensors, as done in [here](https://quimb.readthedocs.io/en/latest/tensor-1d.html)
- if possible, can you describe relevant part of https://journals.aps.org/pra/abstract/10.1103/PhysRevA.76.052315 ? perhaps too much ask, so please go ahead if you're already familiar with this. if not, add a sentence where in the paper the reader should refer to.

Reply:
I added a few more sentences according to your proposal, as well as an additional plot for better visualization of tensors.